packages/theme/src/cli/services/metafields-pull.ts

```diff
@@ -161,5 +161,31 @@ function writeMetafieldDefinitionsToFile(path: string, content: unknown) {
   mkdirSync(shopifyDirectory)
 
   const filePath = joinPath(shopifyDirectory, 'metafields.json')
-  writeFileSync(filePath, JSON.stringify(content, null, 2))
+  const fileContent = JSON.stringify(content, null, 2)
+
+  if (!fileExistsSync(filePath)) {
+    addToGitIgnore(path, '.shopify')
+  }
+
+  writeFileSync(filePath, fileContent)
+}
+
+function addToGitIgnore(root: string, entry: string) {
+  const gitIgnorePath = joinPath(root, '.gitignore')
+
+  if (!fileExistsSync(gitIgnorePath)) {
+    // When the .gitignore file does not exist, the CLI should not be opinionated about creating it
+    return
+  }
+
+  const gitIgnoreContent = readFileSync(gitIgnorePath)
+
+  if (gitIgnoreContent.includes(entry)) {
+    // The file already existing in the .gitignore
+    return
+  }
+
+  const eol = detectEOL(gitIgnoreContent.toString())
```

Comment:
Should we be a little safer with checking the gitignore files incase it could have similar-looking entries. E.g. `/.shopify`, `.shopify/foo`, etc.

```suggestion
  const gitIgnoreContent = readFileSync(gitIgnorePath)
  const content = gitIgnoreContent.toString()
  const eol = detectEOL(gitIgnoreContent.toString())

  if (content.split(eol).includes(entry)) {
    // The file already existing in the .gitignore
    return
  }
```

Reply:
Excellent call, @aswamy! I've applied that change :)